### PR TITLE
Arrow key navigation for the toolbox [RITAccess]

### PR DESCRIPTION
**Pull request by MaryS94**
Arrow key navigation for the toolbox and flyout from the RIT accessibility team.

### CODE REVIEW

**Comment by carlosperate** [comment]
Please excuse me if I am incorrect, but are you sure you were not trying to submit this to https://github.com/RITAccess/blockly ?

**Comment by MaryS94** [comment]
Actually I am from rit access. Earlier this month we had a virtual meeting
with Neil and he recommended sending a pull request for our toolbox
navigation. So I made a separate branch with just the added functionality.
Sorry for the confusion.

**Comment by carlosperate** [comment]
No, completely my assumption, thanks for clarifying.

File: controls.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+Keyboard controls
+
+Up and down arrow to traverse different categories and blocks.
+Left and right arrows to switch between selecting categories and selecting blocks.
+escape to exit the flyout.
+
+Link to RIT Accessibility Blockly: https://github.com/RITAccess/blockly
```
Comment (carlosperate) [inline controls.txt]: Is this file really required? the usage is very intuitive on its own.

File: core/flyout_nav.js [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+*Unless required by applicable law or agreed to in writing, software
+*distributed under the License is distributed on an "AS IS" BASIS,
+*WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*See the License for the specific language governing permissions and
+*limitations under the License.
+*/
+
+/*
+* File overview: this file handles using the keyboard to navigate the flyout
+*/
```
Comment (carlosperate) [inline core/flyout_nav.js]: All the other js files in blockly generally include the full header with the license as the first thing at the top of the file, might be a good idea to follow that pattern.

File: core/flyout_nav.js [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+    oldLength: 0,    
+    currIndex: 0,    
+    prevIndex: 0,
+    opened: false, 
+    currentFlyoutArr: []
+};
+ 
+/*
+* Handles Keyboard input in the toolbox/flyout
+* @param {event}
```
Comment (carlosperate) [inline core/flyout_nav.js]: It's useful to specify the event type so `@param {KeyboardEvent} e Gen description...`

File: core/flyout_nav.js [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+};
+
+
+
+
+/**
+ * Hide and empty the flyout.
+ */
+Blockly.Flyout.prototype.hide = function() {
+flyoutNav.opened = false;
```
Comment (carlosperate) [inline core/flyout_nav.js]: this line needs to be tabbed

**Comment by carlosperate** [comment]
I'd recommend running the google closure linter to adhere to the code style guidelines. I haven't had a chance to test it in a running fork yet, might be able to do so in the next couple of days.

**Comment by MaryS94** [comment]
Thank you, I have used the linter and corrected the issues you pointed out. I just wanted to point out that I am currently refactoring two of the methods (flyoutNavUp and flyoutNavDown) because I realized they were extremely redundant and had terribly long, inefficient if statements. I will commit a much cleaner version of the file soon.

**Comment by TechplexEngineer** [comment]
Might I suggest that this Pull Request be renamed "Arrow key navigation for the toolbox [RITAccess]"

**Comment by MaryS94** [comment]
The name has been updated, thank you for pointing that out!

**Comment by rachel-fenichel** [comment]
Closing as part of spring cleaning.  Blockly now has keyboard navigation, in the toolbox and the workspace.
